### ISSUE TITLE
Allow selecting individual themes or plugins to scan.

### ISSUE THREAD

**Issue by octalmage** [wpengine/phpcompat#67]
Idea from WordPress.org: 

https://wordpress.org/support/topic/checking-a-single-plugintheme/

**Comment by Pross** [comment]
Something like this?
<img width="490" alt="screen shot 2016-09-26 at 21 29 55" src="https://cloud.githubusercontent.com/assets/265860/18850680/7a7f34ee-8430-11e6-91e4-b594ade1db0c.png">

**Comment by ScottDeLuzio** [comment]
+1 for this feature also. I've found myself installing a plugin, not knowing whether or not it is compatible with a PHP version. It would be nice to scan just that one plugin on a staging site before installing. I don't need to scan the whole site since I've already done that.

**Comment by Julix91** [comment]
Main reason I'd like to do this is also to skip the ones that are too big, since I heard about that in the reviews... 

Shouldn't there be a check on size of the plugin before starting? Is the timeout a readable value? If so couldn't it estimate (size of plugin times pessimistic processing speed times timeout = success or failure estimate). Guess that's a different ticket, but I'm sure it also already exists somewhere.

**Comment by Julix91** [comment]
Okay just tested disabling all except one at a time - and that did work... but I like having some plugins hanging around in disabled state (cause I don't actually use them most of the time, just want to be able to add and remove when I do need them briefly) - so with this solution I have to remember which ones were active before etc. to restore that state after everything was tested. Not great.

I like what @Pross  was suggesting for testing specific ones.

I like @davidwolfpaw 's input for having a list of skippable plugins... user editable of course.

I'll play with the code a little and see if it's something I could help with.

**Comment by Julix91** [comment]
Using the normal bulk features in plugin screen would also be nice (i.e. select the ones you want and then check just those)

**Comment by Julix91** [comment]
In case I don't follow through and actually make a PR and someone else wants to:

In the src folder the `wpephpcompat.php` file has the conditional to skip plugins when "only active" is "yes".

Could create a skip list (i.e. comma separated list [can commas be in plugin names? if so needs to be separated by something else) that gets turned into an array and similar to the lines above if the current plugin is in the list skip...

That's where 

![Screen Shot 2020-02-04 at 1 47 56 PM](https://user-images.githubusercontent.com/12877287/73790218-f9ae5c00-4754-11ea-9ce0-0186e7d2081b.png)

Of course, would need to add the option for storing the list somewhere on the plugin screen - same as where the is active only check gets added.

**Comment by Julix91** [comment]
Added a `skip_list` (see PR #258 ) - but I didn't add the same functionality to the command-line version as I had seen someone suggest in a review...  - would be nice to get some responses, see if this is a problem relevant for other people still.

I'm not really sure how to go about implementing @Pross 's suggestion - but I have seen some plugins do that kind of thing before, adding options like that... though I'm not sure I've seen one plugin ad an option like that to all other plugins (only their own or post types)...

This function here adds the link / scan call to action to the plugin:

`public function filter_plugin_links( $links ) {
		if ( current_user_can( WPEPHPCOMPAT_CAPABILITY ) ) {
			$url = add_query_arg( 'page', WPEPHPCOMPAT_ADMIN_PAGE_SLUG, admin_url( 'tools.php' ) );

			array_unshift( $links, '<a href="' . esc_url( $url ) . '">' . esc_html__( 'Start Scan', 'php-compatibility-checker' ) . '</a>' );
		}

		return $links;
	}`

It's called here:
 
![Screen Shot 2020-02-04 at 3 20 17 PM](https://user-images.githubusercontent.com/12877287/73796372-e05fdc80-4761-11ea-8cc6-2f924d280d7a.png)

That means it seems that to add a link like suggested above we'd at least need to loop through all active plugins to add a filter like that... not sure if timing might be an issue with that - wouldn't want to overwrite any plugin actions which the other plugins had added to themselves...

And that's only part of the problem - this is effectively just a link.
We could easily list out all the plugins on the check plugins page, or make a new one, but then it still needs to call the functionality... and so far I don't see functionality for checking just one plugin anywhere in the code, it's just in that loop.

Wonder if that's why it hadn't been implemented earlier, because some refactoring might be in order to do that properly... hm.
